orchestrator/orchestrator.go

```diff
@@ -381,8 +428,69 @@ func (s *Orchestrator) startExecutionStream(ctx context.Context) error {
 			}
 		}
 		if err := s.rpc.Unsubscribe(context.Background(), subscriber, query); err != nil {
-			s.ErrC <- err
+			log().Error(err)
 		}
 	}()
 	return nil
 }
+
+func (s *Orchestrator) log(proc *process.Process, node *process.Process_Node, parentExec *execution.Execution, event *event.Event, newExecHash hash.Hash, msg string) {
+	log := &Log{
+		ProcessHash:     proc.Hash,
+		NodeKey:         node.Key,
+		NodeType:        node.TypeString(),
+		Msg:             msg,
+		Error:           "",
+		CreatedExecHash: newExecHash,
+	}
+	if event != nil {
+		log.EventHash = event.Hash
+	}
+	if parentExec != nil {
+		log.ParentHash = parentExec.Hash
+	}
+	convertLogToLogrus(log).Info(msg)
+	go s.logs.Pub(log, proc.Hash.String())
+	go s.logs.Pub(log, AllLogTopic)
```

Comment:
this is a bit ugly and I plan to improve it when PR https://github.com/mesg-foundation/engine/pull/1810 is merged as I will replace logrus by the logger of cosmos / tendermint

Reply:
I was thinking to create a listener of the logs in the main / orchestrator command that will display the logs in the std out. Like this the orchestrator package doesn't display any logs by itself.
But I want to see what other package are doing when it comes to logs. I think passing an instance of a logger in dep is the way to go. If I can inject a logger that could also have the pubsub so the API can access it, it will be great.